src/org/FatimaHasT/qamp/zadaci/zadaca7/task2/ElectricCar.java

```diff
@@ -0,0 +1,32 @@
+package org.FatimaHasT.qamp.zadaci.zadaca7.task2;
+
+public class ElectricCar extends Vehicle {
+    private final ElectricEngine engine;
+
+    public ElectricCar(final String model, final double capacity) {
+        super(model);
+        this.engine = new ElectricEngine(capacity);
+    }
+
+    public void accelerate(final double speed) {
+        final double factor = 0.4;
+        final double fuelConsumption = factor * speed;
+        if (this.engine.getCapacity() < fuelConsumption) {
+            System.out.println("Not enough fuel for ElectricCar.");
+        } else {
+            this.engine.decreaseCapacity(fuelConsumption);
+            super.accelerate(speed);
+        }
+    }
+
+    public void decelerate(final double speed) {
+        this.engine.brake(speed);
+        super.decelerate(speed);
+    }
```

Comment:
like for how use method calls here, however this has the bug that if the currentSpeed < speed this will charge the battery more than the actual speed decrease